tests/portfolios/test_portfolio_manager.py

```diff
@@ -0,0 +1,114 @@
+
+        trader.is_enabled = True
+        portfolio_manager.handle_balance_update({})
+        update_portfolio_from_balance_mock.assert_called_once()
+
+
+async def test_handle_balance_update_from_order(backtesting_trader):
+    config, exchange_manager, trader = backtesting_trader
+    portfolio_manager = exchange_manager.exchange_personal_data.portfolio_manager
+
+    trader.simulate = False
+    order = BuyMarketOrder(trader)
+
+    if os.getenv('CYTHON_IGNORE'):
+        return
+
+    with patch.object(portfolio_manager, '_refresh_real_trader_portfolio',
+                      new=AsyncMock()) as _refresh_real_trader_portfolio_mock:
+        with pytest.raises(AssertionError):
+            _refresh_real_trader_portfolio_mock.assert_called_once()
+        await portfolio_manager.handle_balance_update_from_order(order)
+        _refresh_real_trader_portfolio_mock.assert_called_once()
+
+    trader.simulate = True
+    with patch.object(portfolio_manager, '_refresh_simulated_trader_portfolio_from_order',
+                      new=AsyncMock()) as _refresh_simulated_trader_portfolio_from_order_mock:
+        with pytest.raises(AssertionError):
+            _refresh_simulated_trader_portfolio_from_order_mock.assert_called_once()
+        await portfolio_manager.handle_balance_update_from_order(order)
+        _refresh_simulated_trader_portfolio_from_order_mock.assert_called_once()
+
+    trader.is_enabled = False
+    trader.simulate = False
+    assert not await portfolio_manager.handle_balance_update_from_order(order)
+
+
+async def test_refresh_simulated_trader_portfolio_from_order(backtesting_trader):
+    config, exchange_manager, trader = backtesting_trader
+    portfolio_manager = exchange_manager.exchange_personal_data.portfolio_manager
+
+    order = BuyLimitOrder(trader)
+    await order.initialize()
+    with patch.object(portfolio_manager.portfolio, 'update_portfolio_available',
+                      new=Mock()) as update_portfolio_available_mock:
+        with pytest.raises(AssertionError):
+            update_portfolio_available_mock.assert_called_once()
+        portfolio_manager._refresh_simulated_trader_portfolio_from_order(order)
+        update_portfolio_available_mock.assert_called_once()
+
+    order.update(
+        price=random_price(),
+        quantity=random_quantity(),
+        symbol="BTC/USDT"
+    )
+    await order.on_fill(force_fill=True)
+    assert order.is_filled()
+
+    if os.getenv('CYTHON_IGNORE'):
+        return
+    with patch.object(portfolio_manager.portfolio, 'update_portfolio_from_filled_order',
+                      new=Mock()) as update_portfolio_from_filled_order_mock:
+        with pytest.raises(AssertionError):
+            update_portfolio_from_filled_order_mock.assert_called_once()
+        portfolio_manager._refresh_simulated_trader_portfolio_from_order(order)
+        update_portfolio_from_filled_order_mock.assert_called_once()
```

Comment:
should we check portfolio amounts ?

Reply:
I don't think this is appropriate because it's not the role of portfolio manager but we can if we want to test the complete call chain. What do you prefer?

Reply:
Hum thats right, if it's tested somewhere else then it should not be tester here as well